### DISCUSSION TITLE
Draft UI

### DISCUSSION THREAD

**Discussion by charlieikosi** [pinpointltdpng/ODFD-PROJECT#8] [General]
Hi @anapitalai @sylvery

Please have a look at the first [UI draft](https://github.com/charlieikosi/ODFD-PROJECT/tree/research_notes/ui) I created. 

Welcome feedback.

**Comment by charlieikosi** [comment]
Do you think this could be done with React.js? I could be wrong but I had a look at a [youtube video](https://youtu.be/b9eMGE7QtTk) that introduced React.js and built a UI out of it.

**Reply by anapitalai** [reply]
I posted a link to one of traversy media's github content, I bought his udemy course last year and that is the project he did, has all the boilerplate code for UI and backend with a mongdb atlas Database that we can easily tweak. He also uses redux for state management in react. On that  we can easily convert that to an ionic app and port that to android/ios./pwa/desktop using capacitor, capacitor has the apis to interact with phone camera.gps,etc. Either we can follow that build design or use that as a guide. Lets wait for Sylv suggestion.

**Reply by charlieikosi** [reply]
That's great @anapitalai and with your training with react.js, I'm pretty sure this is your expertise. Yes I like the UI and the draft UI I created definitely looks like a react.js for the front-end development. I still need to discuss the draft with you all so next week after easter I'll walk you all through what it all means. Once we're happy with the UI designs, then we shall start talking about the system structure/architecture which you've pointed out.

Thanks!

**Reply by sylvery** [reply]
@charlieikosi this UI flow diagram is really good! Its clear enough and easy to understand.
Two things I see are:
1.  From your question, regarding UI. These days, the UI can be built with anything really. As long as the database backend is separate from the application and the data can be fetched using API endpoints. We can use React, Flutter or even Kotlin (Java) and C#. It really depends on the team's strength and the Project Lead's vision on where s/he wants the project to go. But our aim should be to get the minimum viable product (MVP) out within the given timeline. We can continue to refine and improve the application afterwards taking in user feedback and with big data analysis.
2. I don't see a screen for user to manage his/her account settings. I guess that's something that can be factored in as we go along since we don't have screens also for Make a reservation, Navigate to us, Visit website, etc.
Sorry guys nogat time so em liklik quick run down blo me. Happy Easter lo yutla (@charlieikosi @anapitalai) na please take care as yumi continue lo wok na buildim product blo yumi.
Happy Coding, Geniuses!

**Reply by charlieikosi** [reply]
Excellent pool of ideas coming through here and honestly it's richly rewarding collaborating with you lot. Draft UI is up for critique so feel free to call things out. 

Thanks @sylvery for your points. I share the same sentiments to get that MVP.

I'd also like to acknowledge the teams efforts in using github and its other functions. I'm sure we've used it to a level specific for whatever purpose we needed it and this collaboration will further strengthen our knowledge and to also explore its unique features.

Happy Easter!
